### ISSUE TITLE
Questions about Fault tolerance and Checkpoint

### ISSUE THREAD

**Issue by darrenglow** [intelligent-machine-learning/dlrover#770]
When I test the Fault Tolerance feature in dlrover, I met such confusing situation.

I apply the `examples/pytorch/mnist/elastic_job.yaml` in minikube. Two pods are launched. I want to test the fault tolerance inside the pod. So I `docker exec` into the corresponding container. 

```
root@torch-mnist-edljob-worker-1:/dlrover# ps -ef
UID          PID    PPID  C STIME TTY          TIME CMD
root           1       0  0 07:20 ?        00:00:03 /usr/local/bin/python /usr/local/bin/dlrover-run --network-check --nnodes=2 --nproc_per_node=2 --max_restarts=10 examples
root          79       0  0 07:21 pts/0    00:00:00 /bin/bash
root        3186       1 16 07:48 ?        00:00:42 /usr/local/bin/python -u examples/pytorch/mnist/cnn_train.py --num_epochs 5 --training_data /data/mnist_png/training/ --v ...(too long)
root        3187       1 16 07:48 ?        00:00:43 /usr/local/bin/python -u examples/pytorch/mnist/cnn_train.py --num_epochs 5 --training_data /data/mnist_png/training/ --v ...(too long)
root        4399      79  0 07:53 pts/0    00:00:00 ps -ef
```
I try to kill the worker inside the node. `kill 3186`， and then all the workers restart.

When all the workers restart to train, the `torch-mnist-edljob-worker-0` load the checkpoint, but others don't. Meanwhile, the `torch-edljob-worker-0` just have 60 iters per epoch. This is very confusing. I am wondering if this is something wrong with the fault tolerance or I write something wrong.

```
loss = 0.37226542830467224, step = 0
loss = 0.24916566908359528, step = 20
loss = 0.2664795517921448, step = 40
loss = 0.214645117521286, step = 60
Test model after epoch 4
Test the model ...

Test set: Average loss: 0.1753, Accuracy: 9464/10000 (95%)
```

**Comment by darrenglow** [comment]
PS: Other pods still run 460 steps per epoch. When the `torch-mnist-edljob-worker-0` completes 5 epochs as configured, others don't complete running, and after an internal, all workers restart again and again.

**Comment by workingloong** [comment]
You should mount a shard storage like nas/hdfs into all Pods. Otherwise, only the rank0 saves the checkpoint into its local disk and other workers can not read it.

```
# Note, we need to set the path of a shared file
# system like nas, cpfs or hdfs.
CHEKPOINT_PATH = "./checkpoint.pt"
```

**Comment by darrenglow** [comment]
Wow! Thanks a lot! I ignore the explanation in the file at first, so I didn't change the checkpoint save path. 

I have little knowledge about k8s before and I go to learn the `pv` and `pvc`.  It works well now and totally solves my problem. Thanks!

**Comment by darrenglow** [comment]
Oh, I tried to test the fault tolerance and still met a confusing situation.

After 200 iters, a `model.pt` is saved in a shared place. When I kill a worker in a pod, all workers restart and load the checkpoint. After this, every epoch only contains 20 iters. (The number of iters each epoch should contain is about 220, and the checkpoint is saved at iter 200, so it seems $20=220-200$)

**Comment by workingloong** [comment]
> Oh, I tried to test the fault tolerance and still met a confusing situation.
> 
> After 200 iters, a `model.pt` is saved in a shared place. When I kill a worker in a pod, all workers restart and load the checkpoint. After this, every epoch only contains 20 iters. (The number of iters each epoch should contain is about 220, and the checkpoint is saved at iter 200, so it seems 20=220−200)

Thanks for your feedback. I have fixed the bug in the PR #773. After the PR is merged, you can try it again.

**Comment by workingloong** [comment]
The PR #773 has been merged and I have tested it.
